tools/management/commands/assign_structure_states.py

```diff
@@ -243,21 +236,97 @@ def handle(self, *args, **options):
 
                         # Definitely an inactive state structure When distance is smaller than 13Å
                         if entry[1] < 13*distance_scaling_factor:
-                            struct.state, created = ProteinState.objects.get_or_create(slug="inactive", defaults={'name': "Inactive"})
+                            struct.state = ProteinState.objects.get_or_create(slug="inactive", defaults={'name': "Inactive"})[0]
 
                         # UGLY: knowledge-based hardcoded corrections
                         if entry[0] in hardcoded:
                             structure_state = hardcoded[entry[0]]
-                            struct.state, created = ProteinState.objects.get_or_create(slug=structure_state, defaults={'name': structure_state.capitalize()})
+                            struct.state = ProteinState.objects.get_or_create(slug=structure_state, defaults={'name': structure_state.capitalize()})[0]
 
                         # Save changes
                         struct.save()
 
-        self.logger.info('DONE assiging the "Degree Active" levels and activation states')
+        self.logger.info("DONE assiging the \"Degree Active\" levels and activation states")
+
+        if not options["states_only"]:
+            # Assigning the representative tag for the "best" structure with the same receptor-state combinations
+            self.logger.info("ASSIGNING the \"representative\" tag for unique structure-state complexes")
+
+            # Set the representative state of all GPCR structure to False
+            Structure.objects.filter(protein_conformation__protein__family__slug__startswith="00").update(representative=False)
```

Comment:
I'm not sure I get the filter here, do we have other Structure objects than receptor? So why not just .all()?

Reply:
That’s with the eye on the new structure models. In principle, all structures should be in one table and the specific properties in another table. In that case the receptors and the signaling proteins would be mixed. But in all honesty, I just forgot that the signaling proteins were not yet in there 😉

Reply:
Subconsciously thinking ahead! I like it! :)